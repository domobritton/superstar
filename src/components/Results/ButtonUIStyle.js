import styled from 'styled-components';

import StarRate from '@material-ui/icons/StarRate';

export const Star = styled(StarRate)`
  margin-right: auto;
`;
