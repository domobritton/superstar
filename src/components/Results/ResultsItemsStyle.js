import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const Box = styled(Paper)`
  display: flex;
  padding: 2%;
  text-align: left;
  color: #282c34;
`;

export const Repo = styled.h2`
  margin-top: 0;
  font-size: 22px;
  color: gray;

  @media (max-width: 768px) {
    font-size: 16px;
  }
`;

export const Description = styled.div`
  margin-top: 10px;
  margin-bottom: 20px;
  color: lightgray;
  word-wrap: wrap;

  @media (max-width: 768px) {
    font-size: 12px;
  }
`;

export const Left = styled.div`
  display: flex;
  flex-direction: column;
  width: 90%;

  @media (max-width: 450px) {
    width: 65%;
  }
`;

export const Info = styled.div`
  display: flex;

  @media (max-width: 768px) {
    font-size: 9px;
  }
`;

export const Text = styled.div`
  font-size: 14px;

  @media (max-width: 768px) {
    display: none;
  }
`;

export const Language = styled.div`
  margin: auto 20px;
`;

export const StarCount = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
`;

export const Right = styled.div`
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
`;

export const Github = styled(FontAwesomeIcon)`
  transition: all 0.2s ease-in-out;
  &:hover {
    color: lightgray;
  }
`;

export const Link = styled.a`
  margin-left: auto;
  margin-bottom: 20px;
`;
