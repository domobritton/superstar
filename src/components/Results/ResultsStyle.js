import styled from 'styled-components';

export const Page = styled.div`
  flex-grow: 1;
  padding: 0 5%;
`;

export const Count = styled.div`
  margin-left: 20px;
  font-size: 22px;
  height: 30px;
  color: gray;
  letter-spacing: 6px;
  font-weight: 700;
`;

export const Name = styled(Count)``;
