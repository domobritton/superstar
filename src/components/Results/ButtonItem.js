import React from 'react';
import Button from '@material-ui/core/Button';
import { Star } from './ButtonUIStyle';

export const ButtonItem = ({ data, error, mutateStar, starred, text, id }) => {
  const starColor = starred ? { color: 'gold' } : { color: 'white' };
  return (
    <>
      {error && <div style={{ fontSize: 8 }}>ORG LIMITS ACCESS</div>}
      <Button
        data-testid={'mutate'}
        onClick={e => {
          e.preventDefault();
          mutateStar({
            variables: { input: { starrableId: id } },
          });
        }}
        variant="contained"
        color="primary"
        style={{ width: 120 }}
      >
        <Star style={starColor} />
        {text}
      </Button>
    </>
  );
};
