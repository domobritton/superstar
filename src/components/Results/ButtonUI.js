import React from 'react';
import PropTypes from 'prop-types';
import { withApollo, Mutation } from 'react-apollo';

import { ADD_STAR, REMOVE_STAR } from '../../utilities/graphql/star';

import Button from '@material-ui/core/Button';
import { Star } from './ButtonUIStyle';

const ButtonUI = ({ id, starred }) => {
  const starColor = starred ? { color: 'gold' } : { color: 'white' };
  return (
    <div>
      {starred ? (
        <Mutation mutation={REMOVE_STAR} variable={{ input: id }}>
          {(removeStar, { data }) => (
            <Button
              data-testid={'mutate'}
              onClick={e => {
                e.preventDefault();
                removeStar({
                  variables: { input: { starrableId: id } },
                });
              }}
              variant="contained"
              color="primary"
              style={{ width: 120 }}
            >
              <Star style={starColor} />
              UNSTAR
            </Button>
          )}
        </Mutation>
      ) : (
        <Mutation mutation={ADD_STAR} variable={{ input: id }}>
          {(addStar, { data }) => (
            <Button
              data-testid={'mutate'}
              onClick={e => {
                e.preventDefault();
                addStar({
                  variables: { input: { starrableId: id } },
                });
              }}
              variant="contained"
              color="primary"
              style={{ width: 120 }}
            >
              <Star style={starColor} />
              STAR
            </Button>
          )}
        </Mutation>
      )}
    </div>
  );
};

ButtonUI.propTypes = {
  Id: PropTypes.number,
  starred: PropTypes.bool,
};

export default withApollo(ButtonUI);
