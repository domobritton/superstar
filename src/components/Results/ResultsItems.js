import React from 'react';
import PropTypes from 'prop-types';

import ButtonUI from './ButtonUI';

import Grid from '@material-ui/core/Grid';
import StarRate from '@material-ui/icons/StarRate';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import {
  Box,
  Repo,
  Description,
  Left,
  Right,
  Info,
  Text,
  Language,
  StarCount,
  Github,
  Link,
} from './ResultsItemsStyle';

export const ResultsItems = ({ node }) => {
  return (
    <Grid item xs={12}>
      <Box>
        <Left>
          <Repo>{node.nameWithOwner}</Repo>
          <Description>{node.description}</Description>
          <Info>
            <Text>Language</Text>
            {node.primaryLanguage !== null && (
              <Language style={{ color: node.primaryLanguage.color }}>
                {node.primaryLanguage.name}
              </Language>
            )}
            <StarCount>
              {node.stargazers.totalCount}
              <StarRate />
            </StarCount>
          </Info>
        </Left>
        <Right>
          <Link href={node.url} target="_blank" rel="noopener noreferrer">
            <Github icon={faGithub} size="2x" color="gray" />
          </Link>
          <ButtonUI id={node.id} starred={node.viewerHasStarred} />
        </Right>
      </Box>
    </Grid>
  );
};

ResultsItems.propTypes = {
  node: PropTypes.object,
};
