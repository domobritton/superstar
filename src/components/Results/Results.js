import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withApollo, Query } from 'react-apollo';

import { SEARCH_QUERY } from '../../utilities/graphql/searchQuery';

import { ResultsItems } from './ResultsItems';

import Grid from '@material-ui/core/Grid';
import { Page, Count } from './ResultsStyle';

const Results = ({ search, value }) => {
  if (!search) return null;
  const key = value === 1 ? 'user' : 'repository';
  return (
    <Page>
      <Query
        query={SEARCH_QUERY}
        variables={{ queryString: `${key}:${search}` }}
      >
        {({ data, loading, error }) => {
          if (loading) return 'Loading...';
          if (error) return `Error! ${error.message}`;
          return (
            <Grid container spacing={24}>
              <Count>{data.search.repositoryCount} REPOSITORIES FOUND</Count>
              {data.search.edges.map(({ node }) => (
                <Fragment key={node.id}>
                  <ResultsItems node={node} />
                </Fragment>
              ))}
            </Grid>
          );
        }}
      </Query>
    </Page>
  );
};

Results.propTypes = {
  search: PropTypes.string,
  value: PropTypes.number,
};

export default withApollo(Results);
