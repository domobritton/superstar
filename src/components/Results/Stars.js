import React, { Fragment } from 'react';
import { withApollo, Query } from 'react-apollo';

import { STARS_QUERY } from '../../utilities/graphql/starsQuery';

import { ResultsItems } from './ResultsItems';

import Grid from '@material-ui/core/Grid';
import { Page, Name } from './ResultsStyle';

const Stars = () => (
  <Page>
    <Query query={STARS_QUERY}>
      {({ data, loading, error }) => {
        if (loading) return 'Loading...';
        if (error) return `Error! ${error.message}`;
        return (
          <Grid container spacing={24}>
            <Name>{data.viewer.name}</Name>
            {data.viewer.starredRepositories.edges.map(({ node }) => (
              <Fragment key={node.id}>
                <ResultsItems node={node} />
              </Fragment>
            ))}
          </Grid>
        );
      }}
    </Query>
  </Page>
);

export default withApollo(Stars);
