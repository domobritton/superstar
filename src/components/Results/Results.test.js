import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ApolloProvider } from 'react-apollo';
import Results from './Results';

afterEach(cleanup);

const clientMock = jest.fn();

test('<Results />', () => {
  const { container } = render(
    <ApolloProvider client={{}}>
      <Results />
    </ApolloProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
