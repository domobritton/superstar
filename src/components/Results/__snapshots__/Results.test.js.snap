// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> 1`] = `null`;
